src/test/regress/expected/grant_on_database_propagation.out

```diff
@@ -0,0 +1,712 @@
+-- Public role has connect,temp,temporary privileges on database
+-- To test these scenarios, we need to revoke these privileges from public role
+-- since public role privileges are inherited by new roles/users
+revoke connect,temp,temporary  on database regression from public;
+CREATE SCHEMA grant_on_database_propagation;
+SET search_path TO grant_on_database_propagation;
+-- test grant/revoke CREATE privilege propagation on database
+create user myuser;
+grant create on database regression to myuser;
+set role myuser;
+select has_database_privilege('regression', 'CREATE');
+ has_database_privilege
+---------------------------------------------------------------------
+ t
+(1 row)
+
+\c - - - :worker_1_port;
```

Comment:
Afaik this reconnect loses the `set role myuser`, that you did on the first connection. So the next statement isn't testing what you want. I think it's better to use the three argument version of `has_database_privelege` instead of using `SET ROLE` to avoid mistakes in the test like this: `has_database_privilege('myuser', 'regression', 'create')`

Reply:
To be clear, it's not weird that you forgot. I would probably forget myself too in one of the places. Which is why I'm suggesting we use the 3 parameter version, so we don't have to be so careful.